lib/scholar/metrics/discounted_cumulative_gain.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Scholar.Metrics.DiscountedCumulativeGain do
+  @moduledoc """
+  Discounted Cumulative Gain (DCG) is a measure of ranking quality.
+  It is based on the assumption that highly relevant documents appearing lower
+  in a search result list should be penalized as the graded relevance value is
+  reduced logarithmically proportional to the position of the result.
+  """
+
+  @doc """
+  Computes the DCG based on true relevance scores (`y_true`) and their respective predicted scores (`y_score`).
+  """
+  def compute(y_true, y_score, k \\ nil) do
```

Comment:
We want to pass an additional argument via a keyword list named `opts`. Look at the other files in Scholar how it is done. Apart from this, we validate our opts schema using `NimbleOptions` :)